### PR TITLE
Us003 view capability

### PR DESCRIPTION
**Pull request by ConnMeehan**

### CODE REVIEW

**Review by philipwilson5 — COMMENTED** [review]
Build failing

**Review by AlexMcC02 — APPROVED** [review]
Looks good to me.

**Review by philipwilson5 — APPROVED** [review]
Looks good!

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -27,12 +27,12 @@ jobs:
...
-      run: mvn -B package --file pom.xml
+      - uses: actions/checkout@v3
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'temurin'
+          cache: maven
+      - name: Build with Maven
+        run: mvn -B package --file pom.xml
```
Comment (shaunganley) [inline .github/workflows/maven.yml]: these indent changes have broken the tests running, please revert

File: src/main/java/org/kainos/ea/controller/JobRoleController.java [hunk 2]
```diff
@@ -35,4 +34,18 @@ public Response getAllJobRoles() {
...
+    public Response getCapability() {
+
+        try {
+            return Response.ok(jobRoleService.getCapability()).build();
+        } catch (FailedToGetCapabilityException e) {
+            System.err.println(e.getMessage());
+
+            return Response.serverError().build();
+        }
+    }
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/controller/JobRoleController.java]: you don't need a new endpoint, you can just return the capability as part of the existing get roles endpoint for US001

File: src/main/java/org/kainos/ea/dao/JobRoleDao.java [hunk 3]
```diff
@@ -31,4 +30,21 @@ public List<JobRole> getAllJobRoles(Connection c) throws SQLException {
...
+        while (rs.next()) {
+            Capability capability = new Capability(
+                    rs.getString("Role Name"),
+                    rs.getString("Capability")
+            );
+            capabilityList.add(capability);
+        }
+        return capabilityList;
+    }
+}
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/dao/JobRoleDao.java]: this isn't needed, update the existing query to return the capability

File: src/main/java/org/kainos/ea/exception/FailedToGetCapabilityException.java [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package org.kainos.ea.exception;
+
+public class FailedToGetCapabilityException extends Throwable {
+    @Override
+    public String getMessage() {
+        return "Failed to get capability.";
+    }
+}
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/exception/FailedToGetCapabilityException.java]: not needed

File: src/main/java/org/kainos/ea/model/Capability.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    }
+
+    public String getCapability() {
+        return capability;
+    }
+
+    public void setCapability(String capability) {
+        this.capability = capability;
+    }
+}
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/model/Capability.java]: not needed

File: src/main/java/org/kainos/ea/service/JobRoleService.java [hunk 6]
```diff
@@ -27,4 +29,12 @@ public List<JobRole> getAllJobRoles() throws FailedToGetJobRolesException {
...
-}
+    public List<Capability> getCapability() throws FailedToGetCapabilityException {
+        try {
+            return jobRoleDao.getCapability(databaseConnector.getConnection());
+        } catch (SQLException | DatabaseConnectionException e) {
+            System.err.println(e.getMessage());
+            throw new FailedToGetCapabilityException();
+        }
+    }
+}
```
Comment (shaunganley) [inline src/main/java/org/kainos/ea/service/JobRoleService.java]: not needed

**Review by shaunganley — CHANGES_REQUESTED** [review]
seems to have been some misunderstanding with what was required for this ticket. rather than creating a new endpoint, update the existing endpoint to return the capability

**Review by shaunganley — COMMENTED** [review]
looks good, can you update an integration test to check that the capability is being returned and is not null?

**Review by shaunganley — APPROVED** [review]

**Pull request merged**